pkg/platform/kube/test/platform_test.go

```diff
@@ -0,0 +1,195 @@
+			Name:      createFunctionOptions.FunctionConfig.Meta.Name,
+			Namespace: createFunctionOptions.FunctionConfig.Meta.Namespace,
+		})
+		suite.Require().NoError(err, "Failed to get functions")
+
+		function := functions[0]
+		suite.Require().NotEqual(resourceVersion,
+			function.GetConfig().Meta.ResourceVersion,
+			"Resource version should be changed between deployments")
+
+		// we expect a failure due to a stale resource version
+		suite.DeployFunctionExpectError(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
+			suite.Require().Nil(deployResult, "Deployment results is nil when creation failed")
+			return true
+		})
+
+		return true
+	}
+
+	suite.DeployFunctionAndRedeploy(createFunctionOptions, afterFirstDeploy, afterSecondDeploy)
+}
+
+func (suite *DeployFunctionTestSuite) TestAugmentedConfig() {
+	runAsUserID := int64(1000)
+	runAsGroupID := int64(2000)
+	functionAvatar := "demo-avatar"
+	functionLabels := map[string]string{
+		"my-function": "is-labeled",
+	}
+	suite.PlatformConfiguration.FunctionAugmentedConfigs = []platformconfig.LabelSelectorAndConfig{
+		{
+			LabelSelector: metav1.LabelSelector{
+				MatchLabels: functionLabels,
+			},
+			FunctionConfig: functionconfig.Config{
+				Spec: functionconfig.Spec{
+					Avatar: functionAvatar,
+				},
+			},
+			Kubernetes: platformconfig.Kubernetes{
+				Deployment: &appsv1.Deployment{
+					Spec: appsv1.DeploymentSpec{
+						Template: v1.PodTemplateSpec{
+							Spec: v1.PodSpec{
+								SecurityContext: &v1.PodSecurityContext{
+									RunAsUser:  &runAsUserID,
+									RunAsGroup: &runAsGroupID,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	functionName := "augmented-config"
+	createFunctionOptions := suite.compileCreateFunctionOptions(functionName)
+	createFunctionOptions.FunctionConfig.Meta.Labels = functionLabels
+	suite.DeployFunction(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
+		deploymentInstance := &appsv1.Deployment{}
+		functionInstance := &nuclioio.NuclioFunction{}
+		suite.getResourceAndUnmarshal("nucliofunction",
+			functionName,
+			functionInstance)
+		suite.getResourceAndUnmarshal("deployment",
+			kube.DeploymentNameFromFunctionName(functionName),
+			deploymentInstance)
+
+		// ensure function spec was enriched
+		suite.Require().Equal(functionAvatar, functionInstance.Spec.Avatar)
+
+		// ensure function deployment was enriched
+		suite.Require().NotNil(deploymentInstance.Spec.Template.Spec.SecurityContext.RunAsUser)
+		suite.Require().NotNil(deploymentInstance.Spec.Template.Spec.SecurityContext.RunAsGroup)
+		suite.Require().Equal(runAsUserID, *deploymentInstance.Spec.Template.Spec.SecurityContext.RunAsUser)
+		suite.Require().Equal(runAsGroupID, *deploymentInstance.Spec.Template.Spec.SecurityContext.RunAsGroup)
+		return true
+	})
+}
+
+func (suite *DeployFunctionTestSuite) TestMinMaxReplicas() {
```

Comment:
```suggestion
func (suite *DeployFunctionTestSuite) TestHPASpec() {
```

MinMaxReplicas indicate something a bit different about what's actually being deployed, and behavior of autoscale